physics/discrete_trajectory_body.hpp

```diff
@@ -638,12 +663,7 @@ void DiscreteTrajectory<Frame>::FillSubTreeFromMessage(
 
 template<typename Frame>
 Hermite3<Instant, Position<Frame>> DiscreteTrajectory<Frame>::GetInterpolation(
-    Instant const& time) const {
-  CHECK_LE(t_min(), time);
-  CHECK_GE(t_max(), time);
-  // This is the upper bound of the interval upon which we will do the
-  // interpolation.
-  auto const upper = this->LowerBound(time);
+    Iterator const& upper) const {
   auto const lower = upper == this->begin() ? upper : --Iterator{upper};
```

Comment:
I believe that this function is never called with `this->begin()` anymore.  I would put a `CHECK` to that effect and simplify the computation of `lower`.